torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3643,13 +3639,8 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   // to wait() on the returned handle, so ProcessGroupNCCL can't know
   // when it's safe to release the input back to the allocator,
   // and the present call has no way to know it's not an isend.
-  // Therefore, we warn and fall back to the typical recordStream logic:
-  if (avoidRecordStreams_) {
-    TORCH_WARN_ONCE(
-        "TORCH_NCCL_AVOID_RECORD_STREAMS=1 has no effect for point-to-point "
-        "collectives.");
-  }
-
+  // Therefore, we warn and fall back to the typical recordStream logic.
+  // TODO( kwen2501 ): revisit this when we have a better solution.
```

Comment:
I can work on a follow-up with the solution discussed (do `wait()` in destruction if stashed tensors are alive)

Reply:
Actually looking at the existing code 
```
  // Both send tensor and recv tensor are created on a worker stream and used
  // in different ncclStreams.  Hence, both must record the ncclStream to
  // prevent being freed before the collective finishes.
  //
  // See [Sync Streams].
  c10::cuda::CUDACachingAllocator::recordStream(
      tensor.storage().data_ptr(), ncclStream);
 ```
 I'm wondering if this should only be done in the non-batched case. In the batched case and we have `coalescing_state_ == true`, is this redundant as either the `recordStream`/stashing would happen in `endCoalescing`, right?

Reply:
The `pointToPoint` function is basically `isend` and `irecv`, so it is always async. So here we should do what we do for async -- stash the input/output tensor onto CPU stage. Regardless of single P2P or batch P2P.

Re `endCoalescing`, I don't think it does anything related to stash/unstash. The unstash for P2P happens in `work.wait` written by user, applicable to both single P2P and for batched P2P.

Let me know if there is anything unclear :)